src/theme/BlogSidebar/Mobile/index.js

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import Link from '@docusaurus/Link';
+import {NavbarSecondaryMenuFiller} from '@docusaurus/theme-common';
+import SearchBar from "../../SearchBar";
+import styles from './styles.module.css'
+
+function BlogSidebarMobileSecondaryMenu({sidebar}) {
+  return (
+    <>
+      <div className={styles.searchBarContainer}>
+        <SearchBar/>
+      </div>
+      <ul className="menu__list">
+        {sidebar.items.map((item) => (
+          <li key={item.permalink} className="menu__list-item">
+            <Link
+              isNavLink
+              to={item.permalink}
+              className="menu__link"
+              activeClassName="menu__link--active">
+              {item.title}
+            </Link>
+          </li>
+        ))}
+      </ul>
+    </>
+  );
+}
+
+export default function BlogSidebarMobile(props) {
```

Comment:
What's the reason to create separate components? One for Desktop and one for mobile? Why not merging them into one component? Is this the common solution for supporting mobile?

Reply:
This is exported from the docusaurus side
We use sizzle to get the code from docusaurus end and modify it the rest would be automatically added from the original end
https://docusaurus.io/docs/swizzling